packages/@aws-cdk/aws-s3/test/test.auto-delete-objects.ts

```diff
@@ -0,0 +1,218 @@
+import { expect, haveResource } from '@aws-cdk/assert';
+import * as cdk from '@aws-cdk/core';
+import { Test } from 'nodeunit';
+import * as sinon from 'sinon';
+import * as s3 from '../lib';
+import { handler } from '../lib/auto-delete-objects-handler/index';
+
+export = {
+  'when autoDeleteObjects is enabled, a custom resource is provisioned + a lambda handler for it'(test: Test) {
```

Comment:
Could we try and use `jest` for unit tests and its mocking features like `jest.mock` and `jest.spyOn` etc.? I think we anyway prefer to use `jest` for any new change.

Reply:
You can checkout [this](https://stackoverflow.com/a/61863825) example from stack overflow.

Reply:
Thanks for the review :D Yeah I didn't realize that the cdk is trying to move tests over to Jest, so I think that's a good call.

One small issue is that if I try enabling jest for just this one test, I get an error:

> Error: Jest is enabled, but 9 nodeunit tests were found!: test.aspect.js,test.auto-delete-objects.js,test.bucket-policy.js,test.bucket.js,test.cors.js,test.metrics.js,test.notification.js,test.rules.js,test.util.js

I've gone ahead and opened #10934 to unblock this by converting all of the other tests over, so once that is approved I can convert the new test to jest so sinon won't be needed. 👍